intake_geopandas/geopandas.py

```diff
@@ -60,25 +68,62 @@ def __init__(self, urlpath, bbox=None,
             opened. Some examples:
             - ``{{ CATALOG_DIR }}data/states.shp``
             - ``http://some.domain.com/data/states.geo.json``
+
+        use_fsspec: bool
+            Whether to use fsspec to open `urlpath`. By default, `urlpath` is passed
+            directly to GeoPandas, which opens the file using `fiona`. However, for some
+            use cases it may be beneficial to read the file using `fsspec` before
+            passing the resulting bytes to GeoPandas (e.g., when using `fsspec` caching).
+            Note that fiona/GDAL and `fsspec` have mutually-incompatible URL chaining
+            syntaxes, so the URLs passed to each may be significantly different.
+
+        storage_options: dict
+            Storage options to pass to fsspec when opening. Only used when
+            `use_fsspec=True`.
+
         bbox : tuple | GeoDataFrame or GeoSeries, default None
             Filter features by given bounding box, GeoSeries, or GeoDataFrame.
             CRS mis-matches are resolved if given a GeoSeries or GeoDataFrame.
+
         geopandas_kwargs : dict
             Any further arguments to pass to geopandas's read_file function.
         """
         self.urlpath = urlpath
+        self._use_fsspec = use_fsspec
+        self._storage_options = storage_options or {}
         self._bbox = bbox
         self._geopandas_kwargs = geopandas_kwargs or {}
         self._dataframe = None
+        self.storage_options = storage_options or {}
 
         super().__init__(metadata=metadata)
 
     def _open_dataset(self):
         """
-        Open dataset using geopandas and use pattern fields to set new columns.
+        Open dataset using geopandas.
         """
-        self._dataframe = geopandas.read_file(
-            self.urlpath, bbox=self._bbox, **self._geopandas_kwargs)
+        if self._use_fsspec:
+            with fsspec.open_files(self.urlpath, **self._storage_options) as f:
+                f = self._resolve_single_file(f) if len(f) > 1 else f[0]
+                self._dataframe = geopandas.read_file(
+                    f,
+                    bbox=self._bbox,
+                    **self._geopandas_kwargs,
+                )
+        else:
+            self._dataframe = geopandas.read_file(
+                self.urlpath,
+                bbox=self._bbox,
+                **self._geopandas_kwargs
+            )
+
+    def _resolve_single_file(self, filelist):
+        """
+        Given a list of fsspec OpenFiles, choose one to pass to geopandas.
+        """
+        raise NotImplementedError(
```

Comment:
To Do

Reply:
I don't think it's a a TODO, unless there are other multi-file formats we want to support than shapefile.